pkg/component/dependencywatchdog/access.go

```diff
@@ -67,42 +74,77 @@ type dependencyWatchdogAccess struct {
 
 // AccessValues contains configurations for the component.
 type AccessValues struct {
-	// ServerOutOfCluster is the out-of-cluster address of a kube-apiserver.
-	ServerOutOfCluster string
 	// ServerInCluster is the in-cluster address of a kube-apiserver.
 	ServerInCluster string
 }
 
 func (d *dependencyWatchdogAccess) Deploy(ctx context.Context) error {
+	// Create shoot access secret
 	caSecret, found := d.secretsManager.Get(v1beta1constants.SecretNameCACluster)
 	if !found {
 		return fmt.Errorf("secret %q not found", v1beta1constants.SecretNameCACluster)
 	}
 
-	for name, server := range map[string]string{
-		InternalProbeSecretName: d.values.ServerInCluster,
-		ExternalProbeSecretName: d.values.ServerOutOfCluster,
-	} {
-		var (
-			shootAccessSecret = gardenerutils.NewShootAccessSecret(name, d.namespace).WithNameOverride(name)
-			kubeconfig        = kubernetesutils.NewKubeconfig(
-				d.namespace,
-				clientcmdv1.Cluster{Server: server, CertificateAuthorityData: caSecret.Data[secretsutils.DataKeyCertificateBundle]},
-				clientcmdv1.AuthInfo{Token: ""},
-			)
+	var (
+		shootAccessSecret = gardenerutils.NewShootAccessSecret(KubeConfigSecretName, d.namespace).WithNameOverride(KubeConfigSecretName)
+		kubeconfig        = kubernetesutils.NewKubeconfig(
+			d.namespace,
+			clientcmdv1.Cluster{Server: d.values.ServerInCluster, CertificateAuthorityData: caSecret.Data[secretsutils.DataKeyCertificateBundle]},
+			clientcmdv1.AuthInfo{Token: ""},
 		)
+	)
+
+	if err := shootAccessSecret.WithKubeconfig(kubeconfig).Reconcile(ctx, d.client); err != nil {
+		return err
+	}
+
+	// Create MR
+	var registry = managedresources.NewRegistry(kubernetes.SeedScheme, kubernetes.SeedCodec, kubernetes.SeedSerializer)
```

Comment:
Should be using shootScheme and so on.
```suggestion
	var registry = managedresources.NewRegistry(kubernetes.ShootScheme, kubernetes.ShootCodec, kubernetes.ShootSerializer)
```